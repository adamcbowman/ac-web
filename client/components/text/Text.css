@import '../../styles/colors.css';

.Muted {
    color: var(--gray);
}

.Loading {
    color: var(--gray);
}

.Error {
    color: var(--brand-secondary);
}

.Helper {
    cursor: help;
    border-bottom: 1px dotted var(--gray);
}

.Warning {
    color: #ec971f;
}
