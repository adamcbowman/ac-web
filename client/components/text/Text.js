import React, { createElement } from 'react'
import styles from './Text.css'

export function Muted({ children, ...props }) {
    return (
        <p className={styles.Muted} {...props}>
            {children}
        </p>
    )
}

export function Loading({ children = 'Loading...', show, ...props }) {
    return show === true || show === undefined ? (
        <p className={styles.Loading} {...props}>
            {children}
        </p>
    ) : null
}

export function Error({
    children = 'An error happened.',
    component = 'p',
    ...props
}) {
    Object.assign(props, { className: styles.Error })

    return createElement(component, props, children)
}

export function Helper({ children, ...props }) {
    return (
        <span className={styles.Helper} {...props}>
            {children}
        </span>
    )
}
