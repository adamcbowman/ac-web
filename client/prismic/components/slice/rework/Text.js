import React from 'react'
import PropTypes from 'prop-types'
import { StructuredText } from 'prismic/components/base'

Text.propTypes = {
    primary: PropTypes.shape({
        content: PropTypes.arrayOf(PropTypes.object).isRequired,
    }).isRequired,
}

export default function Text({ primary }) {
    return <StructuredText value={primary.content} />
}
