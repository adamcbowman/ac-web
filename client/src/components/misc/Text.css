@import "../../styles/colors.css";

.Text {
    color: var(--gray);
}

.Loading {
    composes: Text;
}

.Error {
    composes: Text;

    color: var(--brand-secondary);
}
