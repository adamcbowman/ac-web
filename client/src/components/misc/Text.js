import React, { PropTypes, DOM } from 'react'
import {compose, setDisplayName, mapProps} from 'recompose'
import CSSModules from 'react-css-modules'
import styles from './Misc.css'

// TODO: Need to extend to all type of text....see Bootstrap for example
// TODO: Show should be removed...and perhaps replace with hide !!!!

Text.propTypes = {
    children: PropTypes.node,
    show: PropTypes.bool,
}

function Text({show = false, children}) {
    if (!show) {
        return null
    }

    return (
        <p styleName='Text'>
            {children}
        </p>
    )
}

Text = CSSModules(Text, styles)

export default Text

export function createText(displayName, defaultText) {
    return compose(
        setDisplayName(displayName),
        mapProps(({children, ...rest}) => ({
            ...rest,
            children: children || defaultText
        }))
    )(Text)
}
