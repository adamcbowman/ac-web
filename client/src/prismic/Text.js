import React, { PropTypes, createElement, DOM } from 'react'

Text.propTypes = {
    document: PropTypes.object.isRequired,
    fragment: PropTypes.string.isRequired,
    component: PropTypes.oneOfType([PropTypes.string, PropTypes.node])
}

export default function Text({ document, fragment, component = DOM.p }) {
	const frag = document.get(fragment)

	if (frag === null) {
		return null
	}

	return createElement(component, null, frag.asText())
}
