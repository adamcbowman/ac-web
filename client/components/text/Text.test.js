import React from 'react'
import Renderer from 'react-test-renderer'
import { Muted, Loading, Error, Helper } from './Text'

test('text component', () => {
    const muted = Renderer.create(<Muted>Content</Muted>)
    const loading = Renderer.create(<Loading>Content</Loading>)
    const error = Renderer.create(<Error>Content</Error>)
    const helper = Renderer.create(<Helper>Content</Helper>)

    expect(muted).toMatchSnapshot()
    expect(loading).toMatchSnapshot()
    expect(error).toMatchSnapshot()
    expect(helper).toMatchSnapshot()
})
