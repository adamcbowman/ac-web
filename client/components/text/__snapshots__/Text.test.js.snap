// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text component 1`] = `
<p
  className="Muted"
>
  Content
</p>
`;

exports[`text component 2`] = `
<p
  className="Loading"
>
  Content
</p>
`;

exports[`text component 3`] = `
<p
  className="Error"
>
  Content
</p>
`;

exports[`text component 4`] = `
<span
  className="Helper"
>
  Content
</span>
`;
